commands/utils/utils_test.go

```diff
@@ -46,41 +49,168 @@ func TestChdirErr(t *testing.T) {
 
 func TestReportUsage(t *testing.T) {
 	const commandName = "test-command"
-	server := httptest.NewServer(createUsageHandler(t, commandName))
+	server := httptest.NewServer(createUsageHandler(t, commandName, "", ""))
 	defer server.Close()
 
 	serverDetails := &config.ServerDetails{ArtifactoryUrl: server.URL + "/"}
+	var usageGroup sync.WaitGroup
+	usageGroup.Add(1)
 	channel := make(chan error)
-	go ReportUsage(commandName, serverDetails, channel)
+	go func() {
+		channel <- ReportUsage(commandName, serverDetails, &usageGroup)
+	}()
+	usageGroup.Wait()
 	assert.NoError(t, <-channel)
 }
 
 func TestReportUsageError(t *testing.T) {
+	var usageGroup sync.WaitGroup
+	usageGroup.Add(1)
+	channel := make(chan error)
+	go func() {
+		channel <- ReportUsage("", &config.ServerDetails{}, &usageGroup)
+	}()
+	usageGroup.Wait()
+	assert.NoError(t, <-channel)
+
+	usageGroup.Add(1)
+	channel = make(chan error)
+	go func() {
+		channel <- ReportUsage("", &config.ServerDetails{ArtifactoryUrl: "http://httpbin.org/status/404"}, &usageGroup)
```

Comment:
httpbin is not stable. We may get flaky tests.
Let's create a new usage handler that returns 404.